src/core_editor/editor.rs

```diff
@@ -431,7 +396,12 @@ mod test {
                 LineBuffer::from("ab "),
                 LineBuffer::from("ab c")
             ],
-            editor.edits
+            editor
+                .edit_stack
+                .edits()
+                .into_iter()
+                .cloned()
+                .collect::<Vec<_>>()
```

Comment:
I am not that fond of that existing test as it doesn't test the behavior we care about and instead the implementation. More valuable would be to see that the correct line buffer state results from doing the actual undo ops

Reply:
Was trying to keep things as they were. Now that you mention this I will add some more tests to test the logic. Will not track cursor behaviour in these new tests as I plan to add that with the PR that will fix #193 .